packages/xo-web/src/xo-app/dashboard/health/index.js

```diff
@@ -506,7 +505,14 @@ const HANDLED_VDI_TYPES = new Set(['system', 'user', 'ephemeral'])
     .sort()
   const getUserSrs = getSrs.filter([isSrWritable])
   const getAlertMessages = createGetObjectsOfType('message').filter([message => message.name === 'ALARM'])
-  const getVifsByMac = createGetObjectsOfType('VIF').groupBy('MAC')
+  const getVifsByMac = createGetObjectsOfType('VIF')
```

Comment:
We can check if the VIF's VM is a snapshot instead, which will reduce computations

```js
createGetObjectsOfType('VIF').filter(createSelector(getVmSnapshots, vmSnapshots => vif => vmSnapshots[vif.$VM] === undefined))
```

Reply:
We can also use the VMs' VIFs to immediately `pick` the right VIF objects:
```js
createGetObjectsOfType('VIF').pick(
  createCollectionWrapper(createSelector(
    getVms,
    vms => flatMap(vms, 'VIFs').sort()
  ))
)
```
This avoids a loop on the VIFs when one VIF *or* one snapshot changes. What do you think?

Reply:
I don't know the gain of performance because there is 2 loops in your suggestion. But, I don't have any oppositions to not keeping your suggestion.

Reply:
Where's the 2nd loop? Do you mean the `sort`? I was mostly trying to avoid triggering selectors when a VIF *or* a snapshot changes. Mine is only triggered when a VM changes, otherwise it's just a `pick`.